java/src/main/java/com/databricks/mlflow/LoaderModule.java

```diff
@@ -0,0 +1,92 @@
+package com.databricks.mlflow;
+
+import com.databricks.mlflow.models.Model;
+import com.databricks.mlflow.sagemaker.Predictor;
+import com.databricks.mlflow.sagemaker.PredictorLoadingException;
+import com.databricks.mlflow.Flavor;
+
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * A generic loader for encapsulating flavor-specific model deserialization logic.
+ * By extending {@link com.databricks.mlflow.LoaderModule}, models of a specific flavor
+ * can be loaded as generic {@link com.databricks.mlflow.sagemaker.Predictor} objects.
+ * This allows tools, such as model containers, to use the models for inference
+ */
+public abstract class LoaderModule<T extends Flavor> {
+    /**
+     * Loads an MLFlow model as a generic predictor that can be used for
+     * inference
+     *
+     * @throws {@link com.databricks.mlflow.sagemaker.PredictorLoadingException} if
+     * failures are encountered while attempting to load the model from the specified configuration.
```

Comment:
Maybe make the message more concise -- "for any failure encountered while attempting to load the model"